### PR TITLE
Give storyboards a BeatmapInfo to reduce weird method calls

### PR DESCRIPTION
**Pull request by peppy**
Also removes unnecessary background texture (the actual storyboard background spec wasn't implemented correctly anyway).

### CODE REVIEW

File: osu.Game/Storyboards/Drawables/DrawableStoryboard.cs [hunk 1]
```diff
@@ -52,11 +44,6 @@ public DrawableStoryboard(Storyboard storyboard)
             Anchor = Anchor.Centre;
             Origin = Anchor.Centre;
 
-            AddInternal(background = new Background
-            {
-                Anchor = Anchor.Centre,
-                Origin = Anchor.Centre,
-            });
```
Comment (smoogipoo) [inline osu.Game/Storyboards/Drawables/DrawableStoryboard.cs]: Are you sure about removing this? Why?

File: osu.Game/Beatmaps/BeatmapManager_WorkingBeatmap.cs [hunk 2]
```diff
@@ -75,23 +75,32 @@ protected override Track GetTrack()
...
+                Storyboard storyboard;
                 try
                 {
                     using (var beatmap = new StreamReader(store.GetStream(getPathForFile(BeatmapInfo.Path))))
                     {
                         Decoder decoder = Decoder.GetDecoder(beatmap);
 
-                        if (BeatmapSetInfo?.StoryboardFile == null)
-                            return decoder.GetStoryboardDecoder().DecodeStoryboard(beatmap);
+                        // todo: support loading from both set-wide storyboard *and* baetmap specific.
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/BeatmapManager_WorkingBeatmap.cs]: `beatmap` typo

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Storyboards/Drawables/DrawableStoryboard.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Storyboards/Drawables/DrawableStoryboard.cs]: check the commit message / code. as far as i can tell, it's not implemented correctly and has no reason to exist here.

Comment (smoogipoo) [reply osu.Game/Storyboards/Drawables/DrawableStoryboard.cs]: Can you make an issue for this then? Perhaps even with a note of why you believe it's not implemented correctly?

Comment (peppy) [reply osu.Game/Storyboards/Drawables/DrawableStoryboard.cs]: There's many other storyboard features not implemented yet, and I'm not sure if we want to add support for them or not. Am I going to have to create issues for each missing thing? I don't think it's necessary at this point.

Comment (smoogipoo) [reply osu.Game/Storyboards/Drawables/DrawableStoryboard.cs]: Wait what other storyboard features? As far as I'm aware triggers are the only things that aren't implemented.

Comment (smoogipoo) [reply osu.Game/Storyboards/Drawables/DrawableStoryboard.cs]: (and yes that should have an issue thread of its own)

Comment (peppy) [reply osu.Game/Storyboards/Drawables/DrawableStoryboard.cs]: Colour events, for instance

File: osu.Game/Storyboards/Storyboard.cs [hunk 3]
```diff
@@ -36,28 +38,22 @@ public StoryboardLayer GetLayer(string name)
...
+                if (backgroundPath == null)
+                    return false;
 
-            return GetLayer("Background").Elements.Any(e => e.Path.ToLowerInvariant() == backgroundPath);
+                return GetLayer("Background").Elements.Any(e => e.Path.ToLowerInvariant() == backgroundPath);
+            }
         }
 
-        public float AspectRatio(BeatmapInfo beatmapInfo)
-            => beatmapInfo.WidescreenStoryboard ? 16 / 9f : 4 / 3f;
```
Comment (smoogipoo) [inline osu.Game/Storyboards/Storyboard.cs]: You should make an issue for the re-implementation of this, since this is now broken.

Comment (peppy) [reply osu.Game/Storyboards/Storyboard.cs]: It's still implemented to the same level it was (see `CreateDrawable`)

Comment (smoogipoo) [reply osu.Game/Storyboards/Storyboard.cs]: Obviously I meant for the background

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**